accounts-db/benches/bench_read_only_accounts_cache.rs

```diff
@@ -0,0 +1,208 @@
+                shuffled_row
+            })
+            .collect();
+        for new_row in new_rows {
+            new_pubkeys
+                .push_row(ArrayView::from(new_row.as_slice()))
+                .unwrap();
+        }
+        let new_pubkeys: Vec<Vec<Pubkey>> = new_pubkeys
+            .columns()
+            .into_iter()
+            .map(|column| {
+                // Both `ArrayBase::as_slice` and `ArrayBase::as_mut_slice`
+                // return `None` in this case, so let's just collect the elements.
+                let mut pubkeys_for_thread = column
+                    .into_iter()
+                    .map(|pubkey| pubkey.to_owned())
+                    .collect::<Vec<_>>();
+                pubkeys_for_thread.shuffle(&mut rng);
+                pubkeys_for_thread
+            })
+            .collect();
+
+        // Spawn the reader threads in the background.
+        let stop_reader = Arc::new(AtomicBool::new(false));
+        let reader_handles = (0..*num_readers_writers).map(|i| {
+            let cache = cache.clone();
+            let pubkeys = pubkeys.clone();
+            let stop_reader = stop_reader.clone();
+            Builder::new()
+                .name(format!("reader{i:02}"))
+                .spawn({
+                    move || {
+                        // Continuously read random accounts.
+                        let mut rng = SmallRng::seed_from_u64(i as u64);
+                        while !stop_reader.load(Ordering::Relaxed) {
+                            let pubkey = pubkeys.choose(&mut rng).unwrap();
+                            test::black_box(cache.load(*pubkey, slot));
+                        }
+                    }
+                })
+                .unwrap()
+        });
+
+        // Benchmark reads and writes on a full cache, trigerring eviction on each
+        // write.
+        let slot = 1;
+        group.sample_size(10);
+        group.bench_function(
+            BenchmarkId::new("read_only_accounts_cache_eviction", num_readers_writers),
+            |b| {
+                b.iter(|| {
+                    // Perform the writes.
+                    let writer_handles = (0..*num_readers_writers).map(|i| {
+                        let cache = cache.clone();
+                        let new_pubkeys = new_pubkeys[i].clone();
+
+                        Builder::new()
+                            .name(format!("writer{i:02}"))
+                            .spawn({
+                                move || {
+                                    for pubkey in new_pubkeys {
+                                        cache.store(
+                                            pubkey,
+                                            slot,
+                                            Account {
+                                                lamports: 1,
+                                                // 1 MiB
+                                                data: vec![1; 1024 * 1024],
```

Comment:
I think you want to do these allocations outside the bench, otherwise this is
what takes the most time

Reply:
Also, do they have to be 1MB? This is a contention bug right? The size of
accounts shouldn't matter

Reply:
Good point with the allocation, will move outside the bench.

Regarding the size - the size itself doesn't matter, but a bigger size triggers eviction faster. The default eviction threshold is `410 * 1024 * 1024` (410 MiB). `CACHE_ENTRY_SIZE` is 144. If you're benching without account data, you need 2985529 accounts to fill up the cache. This is the reason why the [currently existing bench](https://github.com/anza-xyz/agave/blob/8c3a6bb6898e61b0962d6fe9a892dd29b11da097/accounts-db/benches/accounts.rs#L126-L181) isn't anywhere close to filling up the cache - it creates 1000 accounts without data. 

I'm open to picking other size which is closer to the average on mnb, I'll think of it today.

Reply:
OK, the average size calculated in the most silly way would be:

```
>>> 265633981325/744678683
356.7095277319762
```

356 bytes, lol. That makes sense, given that the most of accounts on mnb are token accounts. In that case, I'm not sure if we should go by some size like 256 or 512 bytes, or should we just skip the size all together and bench with >2985529 empty accounts. Leaning towards the latter.

Reply:
We should _also_ bench the 0 data case, as that gives us the worst case in terms of [number of things we store]. So I'd say 0 and something reasonable, 512b or 1K.

Reply:
>So I'd say 0 and something reasonable, 512b or 1K.

Done exactly that.